torch/csrc/distributed/rpc/rref_impl.cpp

```diff
@@ -146,8 +172,11 @@ IValue UserRRef::toHere() const {
       *agent,
       agent->getWorkerInfo(ownerId_),
       std::move(msgToSend),
-      true /* forceGradRecording */);
+      true /* forceGradRecording */,
+      timeoutSeconds);
 
+  // TODO: we should ideally be able to interrupt this blocking wait if we check
+  // isTimedOut() and it is true.
```

Comment:
isn't the wait() going to be interrupted by the futureResponse being satisfied? or is the issue that the rref_context needs to handleError, and we simply haven't had the timedout set yet?

(we could probably interpret any thrown exception here if it made sense?)

Reply:
The specific case that I was thinking of is something like, 
1) Say rpc.remote() call is completely dropped on the receiver, (or takes a long time to process)
2) Sender, after rpc.remote(), calls rref.to_here(), thus blocking 
3) We are notified that (1) failed (via a timeout), and mark that future with an error and call `setTimedOut()` on the rref. 
4) However since in (2) we have already started blocking on a (different) future, we should ideally be able to interrupt this with the information we have from (3)

Although, this isn't too big of a deal since the blocking future in this call would eventually time out as well.